### PR TITLE
Anchor input fetches to source action id

### PR DESCRIPTION
**Pull request by werkt**
Provide the specific action id via RequestMetadata which provided an
action input artifact when using remote_download_minimal. This replaces
the unattributable "fetch-remote-inputs" identifier populated for each
input via a nested context.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/actions/FileArtifactValue.java [hunk 1]
```diff
@@ -82,6 +82,8 @@
   // TODO(ulfjack): Throw an exception if it's not a file.
   public abstract long getSize();
 
+  public abstract String getActionId();
```
Comment (meisterT) [inline src/main/java/com/google/devtools/build/lib/actions/FileArtifactValue.java:85]: Consider moving the default implementation (`return "";`) here.

**Comment by meisterT** [comment]
cc @ulfjack

File: src/main/java/com/google/devtools/build/lib/actions/FileArtifactValue.java (hunk 1 above)
Comment (werkt) [reply src/main/java/com/google/devtools/build/lib/actions/FileArtifactValue.java]: Done and commented.

**Comment by meisterT** [comment]
This seems to fail on import: https://buildkite.com/bazel/google-bazel-presubmit/builds/33753#_

I'll have a closer look tomorrow

**Comment by werkt** [comment]
> This seems to fail on import: https://buildkite.com/bazel/google-bazel-presubmit/builds/33753#_

Just needs a merge update. Forthcoming.

**Comment by werkt** [comment]
Ping on this, should be clean for merge.

**Comment by meisterT** [comment]
Sorry George, this needs some changes to internal code. I'll try to fix it next week.

**Comment by coeuvre** [comment]
@werkt We are considering removing `action_id` from remote metadata. Can you elaborate more why would you like to track input prefetches with source `action_id`?

According to the REAPI spec, `action_id` is

>   An identifier that ties multiple requests to the same action. For example, multiple requests to the CAS, Action Cache, and Execution API are used in order to compile foo.cc.

The input prefetches are requested by local actions that depend on outputs of remote executed action (which is identified by the `action_id`). These requests are not part of generating action hence shouldn't be bound with the `action_id`.

cc @tjgq

**Pull request closed without merging**